### ISSUE TITLE
RAPI query interface should support read-only users

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#357]
Originally reported of Google Code with ID 301.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

# gnt-cluster --version
gnt-cluster (ganeti v2.6.0-25-g27e15be) 2.6.1
# gnt-cluster version
Software version: 2.6.1
Internode protocol: 2060000
Configuration format: 2060000
OS api version: 20
Export interface: 0

<b>What distribution are you using?</b>
# cat /etc/*-release
SUSE Linux Enterprise Server 11 (x86_64)
VERSION = 11
PATCHLEVEL = 2

Querieing the RAPI with /2/query/[resource] requires a user with write option set. Hence a query can not write anything, a read-only user should be enough. This is important if you trust someone enough to hand out sensitive information, but not enough to remote control your cluster.

Thanks, Sascha.



```
Originally added on 2012-11-06 14:04:33 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-11-07 14:03:33 +0000 UTC.

Added Labels: Type-Enhancement
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 5e12acf
Author: Michael Hanselmann <hansmi@google.com>
Date:   Thu Nov 8 19:31:14 2012 +0100

    rapi: Add new user option for querying
    
    This was requested in issue 301. Before this patch, requests to
    “/2/query/*” and “/2/instances/*/console” would require authentication
    with a user with write access. Since that is not strictly necessary, a
    new user option named “read” is added.
    
    Console information can also be retrieved as a normal query, therefore
    the change applies there too.
    
    This was the first user option to be added after “write”, therefore
    quite a few changes were necessary. Documentation, including NEWS, is
    updated as well.
    
    Signed-off-by: Michael Hanselmann <hansmi@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>

```
Originally added on 2012-11-16 14:02:52 +0000 UTC.

Mentioned Issue with GoogleCode ID 301 has been migrated to GitHub with ID #357.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
